js/timeout-transition-group.jsx

```diff
@@ -87,6 +84,40 @@ function animationSupported() {
     return endEvents.length !== 0;
 }
 
+/**
+ * Functions for element class management to replace dependency on jQuery
+ * addClass, removeClass and hasClass
+ */
+function addClass(element, className) {
+    if (element.classList) {
+        element.classList.add(className);
+    } else if (!hasClass(element, className)) {
+        element.className = element.className + ' ' + className;
+    }
+    return element;
+}
+function removeClass(element, className) {
+    if (hasClass(className)) {
+        if (element.classList) {
+            element.classList.remove(className);
+        } else {
+            element.className = element.className
+                .replace(className, '')
+                .replace(new RegExp('/(^|\\s)' + className + '($|\\s)/'), ' ')
```

Comment:
I'm fine with using regexes, but if we do we should make sure they are correct and handle all characters that can be in a class name.